pkg/aws/helpers.go

```diff
@@ -249,6 +253,11 @@ func UserTagValidator(input interface{}) error {
 				return fmt.Errorf("invalid tag format, Tags are comma separated, for example: 'foo:bar,bar:baz'")
 			}
 			tag := strings.Split(t, ":")
+
+			if UserTagReservedKeysRE.MatchString(strings.ToLower(tag[0])) {
```

Comment:
I think this would be better if we moved it to the backend. 

If you have a look at - https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/cmd/clusters-service/servecmd/apiserver/validation_helpers.go#L1941

We can also extend this test to capture this new logic there - https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/cmd/clusters-service/servecmd/apiserver/validation_helpers_test.go#L3136

Reply:
WDYT?

Reply:
@ciaranRoche 
Would moving to backend be in addition to keeping in rosa-cli?  So that there is immediate error response in interactive mode?

Reply:
As long as the tags can be parsed into a key value pair, i dont think we need to add additional validations as a immediate error response, this can be handled in the backend and returned to the user. 

If the need for immediate response comes up, we will need to create a third party package to share this logic between CS and ROSA. To avoid maintaining the same logic in both the client and the service.